Kernel/Arch/aarch64/Registers.h

```diff
@@ -487,6 +488,115 @@ struct ESR_EL1 {
+    case 0b011100:
+        return "Exception from a Pointer Authentication instruction authentication failure"sv;
+    case 0b011101:
+        return "Access to SME functionality trapped as a result of CPACR_EL1.SMEN, CPTR_EL2.SMEN, CPTR_EL2.TSM, CPTR_EL3.ESM, or an attempted execution of an instruction that is illegal because of the value of PSTATE.SM or PSTATE.ZA, that is not reported using EC 0b000000"sv;
+    case 0b011110:
+        return "Exception from a Granule Protection Check"sv;
+    case 0b100000:
+        return "Instruction Abort from a lower Exception level"sv;
+    case 0b100001:
+        return "Instruction Abort taken without a change in Exception level"sv;
+    case 0b100010:
+        return "PC alignment fault exception"sv;
+    case 0b100100:
+        return "Data Abort exception from a lower Exception level"sv;
+    case 0b100101:
+        return "Data Abort exception taken without a change in Exception level"sv;
+    case 0b100110:
+        return "SP alignment fault exception"sv;
+    case 0b100111:
+        return "Memory Operation Exception"sv;
+    case 0b101000:
+        return "Trapped floating-point exception taken from AArch32 state"sv;
+    case 0b101100:
+        return "Trapped floating-point exception taken from AArch64 state"sv;
+    case 0b101111:
+        return "SError interrupt"sv;
+    case 0b110000:
+        return "Breakpoint exception from a lower Exception level"sv;
+    case 0b110001:
+        return "Breakpoint exception taken without a change in Exception level"sv;
+    case 0b110010:
+        return "Software Step exception from a lower Exception level"sv;
+    case 0b110011:
+        return "Software Step exception taken without a change in Exception level"sv;
+    case 0b110100:
+        return "Watchpoint exception from a lower Exception level"sv;
+    case 0b110101:
+        return "Watchpoint exception taken without a change in Exception level"sv;
+    case 0b111000:
+        return "BKPT instruction execution in AArch32 state"sv;
+    case 0b111100:
+        return "BRK instruction execution in AArch64 state"sv;
+    default:
+        VERIFY_NOT_REACHED();
+    }
+}
+
+// D17.2.40 FAR_EL1, Fault Address Register (EL1)
+static inline bool exception_class_has_set_far(u8 exception_class)
+{
+    // Faulting Virtual Address for synchronous exceptions taken to EL1. Exceptions that set the
+    // FAR_EL1 are Instruction Aborts (EC 0x20 or 0x21), Data Aborts (EC 0x24 or 0x25), PC alignment
+    // faults (EC 0x22), and Watchpoints (EC 0x34 or 0x35). ESR_EL1.EC holds the EC value for the
+    // exception.
+    return exception_class == 0x20 || exception_class == 0x21 || exception_class == 0x24 || exception_class == 0x25 || exception_class == 0x22 || exception_class == 0x34 || exception_class == 0x35;
```

Comment:
very nit: this probably reads better with a switch?

Reply:
Yeah true, clang-format actually changed it to be on a single line :(

Changed to a switch statement :)